### PR TITLE
Added backwards compatibility for coin_solutions in push_tx

### PR DESCRIPTION
**Pull request by Quexington**

### CODE REVIEW

File: chia/rpc/full_node_rpc_api.py [hunk 1]
```diff
@@ -490,6 +490,9 @@ async def push_tx(self, request: Dict) -> Optional[Dict]:
         if "spend_bundle" not in request:
             raise ValueError("Spend bundle not in request")
 
+        if "coin_solutions" in request["spend_bundle"]:
```
Comment (arvidn) [inline chia/rpc/full_node_rpc_api.py:493]:
```suggestion
        # coin_solutions is the old name, we support it for backwards compatibility
        if "coin_solutions" in request["spend_bundle"]:
```

**Comment by arvidn** [comment]
[this](https://github.com/Chia-Network/chia-blockchain/pull/7574/checks?check_run_id=3088235000#step:14:185) is scary:

```
Fatal Python error: Segmentation fault
```

**Review by arvidn — APPROVED** [review]
I really think we need tests of the public API too. Otherwise it will be very easy to accidentally break it going forward

**Comment by ChiaMineJP** [comment]
Hi @arvidn 
I'm tracking public API changes in order to keep integrity of [my chia client library](https://github.com/Chia-Mine/chia-agent)

You can easily see resulted API changes below.
https://github.com/Chia-Mine/chia-agent/blob/main/CHANGELOG.md

If you provide a way to discuss API changes, I'll post my comment there.

Just FYI.

**Comment by ChiaMineJP** [comment]
@Quexington @arvidn 

Please see the list of APIs affected by the breaking change.
https://github.com/Chia-Network/chia-blockchain/commit/6cf29102f95b410a8c6dc416e612c998bac567fa#comments

**Pull request merged**